ovirt: default save true in setup host networks

erav opened the pull request:
Default value for save updated network configuration
during setup host networks is set to true.

##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->

<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Feature Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
ovirt_host_network

##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->
needs update of the api-model and regeneration of ovirt-sdk4
<!--- Paste verbatim command output below, e.g. before and after your change -->
```paste below

```

erav committed:
ovirt: default save true in setup host networks

Default value for save updated network configuration
during setup host networks is set to true.

erav committed:
ovirt: default save true in setup host networks

Default value for save updated network configuration
during setup host networks is set to true.

erav committed:
ovirt: default save true in setup host networks

Default value for save updated network configuration
during setup host networks is set to true.

dominikholler commented:
Please check the "temporarily" in the doc section.
What happens, if this is module is executed against an old version of the ovirt-sdk, which does not expect the 'save' parameter?

machacekondra commented:
Dominik is right, can you please just change the default value of the `save` parameter and change it's documentation. The save is implemneted in `_action_save_configuration` method. So it will work correctly but will be less effecient as it will do two API calls, but keep the compatibility with 4.2 version of the SDK/API.

erav commented:
posted fix

machacekondra commented:
shipit

machacekondra commented on lib/ansible/modules/cloud/ovirt/ovirt_host_network.py:
```diff
@@ -74,7 +74,7 @@
     save:
         description:
-            - "If I(true) network configuration will be persistent, by default they are temporarily."
+            - "If I(false) network configuration will be temporary until next boot, by default they are persistent."
```
Can you please write that: Since Ansible 2.8 they are persisted by default? So it's clear.

erav committed:
Update ovirt_host_network.py

machacekondra commented:
shipit

mwperina commented on lib/ansible/modules/cloud/ovirt/ovirt_host_network.py:
```diff
@@ -74,7 +74,7 @@
         description:
-            - "If I(true) network configuration will be persistent, by default they are temporarily."
+            - "If I(true) network configuration will be persistent, otherwise it is temporary. Since Ansible 2.8 I(true) is the default."
         type: bool
```
Please add default value also into documentation

erav replied:
Is this not the documentation? Would you please specify where else to make changes?

erav replied:
Is the comment at the start of the file not the documentation? Would you please specify where else to make changes? thanks

mwperina replied:
I think it should be enough just to add `default: True` to the docs, similarly as we have for example in ovirt_vm module: https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/ovirt/ovirt_vm.py#L42

dominikholler reviewed (changes requested):
Please check all occurrences of "temporarily" in the whole file, if they are still valid.

erav committed:
ovirt: default save true in setup host networks

Default value for save updated network configuration
during setup host networks is set to true.

erav committed:
ovirt: default save true in setup host networks

Default value for save updated network configuration
during setup host networks is set to true.

machacekondra commented:
shipit

mwperina reviewed (approved):
shipit

mwperina commented:
Eitan, could you please the PR so it can be merged?
@erav

erav commented:
@mwperina Could I do what please?

machacekondra commented:
rebase

erav commented:
@mwperina

done

On Wed, Feb 27, 2019 at 2:35 PM ansibot <notifications@github.com> wrote:

> @erav <https://github.com/erav> this PR contains the following merge
> commits:
>
>    - b686477
>    <https://github.com/ansible/ansible/commit/b686477ea87eb1dd9337462a14f95f70678f56af>
>
> Please rebase your branch
> <http://docs.ansible.com/ansible/devel/dev_guide/developing_rebasing.html>
> to remove these commits.
>
> click here for bot help
> <https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ansible/ansible/pull/49983#issuecomment-467845477>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEf-CiPpHEtcwem28vsiZUvL8C60c-nmks5vRnuBgaJpZM4ZVNz5>
> .
>

machacekondra commented:
shipit

mwperina commented:
shipit

machacekondra commented:
shipit

mwperina reviewed (approved):
shipit

mwperina commented:
Dominik, you have requested changed in previous review, so will probably need to approve that PR to get it merged automatically

@dominikholler

mwperina commented:
bot_status

machacekondra commented:
shipit

mnecas commented:
shipit

mwperina commented:
shipit
